diff -u dropbear-0.49/runopts.h dropbear-0.49-force-shell/runopts.h
--- dropbear-0.49/runopts.h	2007-02-22 16:51:35.000000000 +0100
+++ dropbear-0.49-force-shell/runopts.h	2007-04-12 22:01:09.000000000 +0200
@@ -85,7 +85,8 @@
 	sign_key *hostkey;
 	buffer * banner;
 	char * pidfile;
-
+    
+    char *force_shell;
 } svr_runopts;
 
 extern svr_runopts svr_opts;
diff -u dropbear-0.49/svr-auth.c dropbear-0.49-force-shell/svr-auth.c
--- dropbear-0.49/svr-auth.c	2007-02-22 16:51:35.000000000 +0100
+++ dropbear-0.49-force-shell/svr-auth.c	2007-04-12 22:07:34.000000000 +0200
@@ -253,6 +253,10 @@
 		/* empty shell in /etc/passwd means /bin/sh according to passwd(5) */
 		usershell = "/bin/sh";
 	}
+    if (svr_opts.force_shell) {
+        usershell = svr_opts.force_shell;
+        goto goodshell;
+    }
 
 	/* check the shell is valid. If /etc/shells doesn't exist, getusershell()
 	 * should return some standard shells like "/bin/sh" and "/bin/csh" (this
diff -u dropbear-0.49/svr-chansession.c dropbear-0.49-force-shell/svr-chansession.c
--- dropbear-0.49/svr-chansession.c	2007-02-22 16:51:35.000000000 +0100
+++ dropbear-0.49-force-shell/svr-chansession.c	2007-04-12 22:09:07.000000000 +0200
@@ -935,6 +935,9 @@
 	} else {
 		usershell = ses.authstate.pw->pw_shell;
 	}
+    if (svr_opts.force_shell) {
+        usershell = svr_opts.force_shell;
+    }
 
 	/* set env vars */
 	addnewvar("USER", ses.authstate.pw->pw_name);
diff -u dropbear-0.49/svr-runopts.c dropbear-0.49-force-shell/svr-runopts.c
--- dropbear-0.49/svr-runopts.c	2007-02-22 16:51:35.000000000 +0100
+++ dropbear-0.49-force-shell/svr-runopts.c	2007-04-12 22:04:29.000000000 +0200
@@ -77,6 +77,8 @@
 					"		(default port is %s if none specified)\n"
 					"-P PidFile	Create pid file PidFile\n"
 					"		(default %s)\n"
+                    "-l shell\n"
+                    "       Always start this shell and ignore shell from /etc/passwd\n"
 #ifdef INETD_MODE
 					"-i		Start for inetd\n"
 #endif
@@ -112,6 +114,7 @@
 	svr_opts.portcount = 0;
 	svr_opts.hostkey = NULL;
 	svr_opts.pidfile = DROPBEAR_PIDFILE;
+    svr_opts.force_shell = NULL;
 #ifdef ENABLE_SVR_LOCALTCPFWD
 	svr_opts.nolocaltcp = 0;
 #endif
@@ -195,6 +198,9 @@
 				case 'P':
 					next = &svr_opts.pidfile;
 					break;
+            case 'l':
+                    next = &svr_opts.force_shell;
+                    break;
 #ifdef DO_MOTD
 				/* motd is displayed by default, -m turns it off */
 				case 'm':
