--- uClibc/utils/ldconfig.c.orig	2008-09-27 16:55:43 +0200
+++ uClibc/utils/ldconfig.c	2008-09-27 16:56:16 +0200
@@ -976,9 +976,9 @@
 		/* look ma, no defaults */
 		if (!nodefault) {
 			scan_dir(UCLIBC_RUNTIME_PREFIX "lib");
-			scan_dir(UCLIBC_RUNTIME_PREFIX "usr/lib");
+			/* scan_dir(UCLIBC_RUNTIME_PREFIX "usr/lib"); */
 #ifndef __LDSO_CACHE_SUPPORT__
-			scan_dir(UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib");
+			/* scan_dir(UCLIBC_RUNTIME_PREFIX "usr/X11R6/lib"); */
 #else
 			/* I guess the defaults aren't good enough */
 			if ((extpath = get_extpath())) {
