--- uClibc/test/string/tester.c.orig	2008-09-25 10:11:20.000000000 +0200
+++ uClibc/test/string/tester.c	2008-09-25 10:14:16.000000000 +0200
@@ -427,6 +427,7 @@
   check (strncmp ("abc", "def", 0) == 0, 13);	/* Zero count. */
   check (strncmp ("abc", "", (size_t)-1) > 0, 14);	/* set sign bit in count */
   check (strncmp ("abc", "abc", (size_t)-2) == 0, 15);
+  check (strncmp ("aa", "ab", (size_t)-1) < 0, 16);
 }
 
 static void
