--- nfs-utils-1.0.10/support/nfs/svc_socket.c.orig	2007-03-22 22:13:20.000000000 +0100
+++ nfs-utils-1.0.10/support/nfs/svc_socket.c	2007-03-22 22:13:57.000000000 +0100
@@ -67,6 +67,7 @@
   __bzero ((char *) &addr, sizeof (addr));
   addr.sin_family = AF_INET;
 
+#ifndef __UCLIBC__
   ret = getrpcbynumber_r (number, &rpcbuf, rpcdata, sizeof rpcdata,
 			  &rpcp);
   if (ret == 0 && rpcp != NULL)
@@ -100,6 +101,7 @@
 	}
     }
   else
+#endif
     {
       if (bindresvport (sock, &addr))
 	{
